src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.sentencing
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+
+@Service
+class SentencingSynchronisationService(
+  private val sentencingMappingService: SentencingMappingService,
+  private val telemetryClient: TelemetryClient
+) {
+  suspend fun synchroniseSentenceAdjustmentCreateOrUpdate(event: SentenceAdjustmentUpsertedOffenderEvent) {
+    sentencingMappingService.findNomisSentencingAdjustmentMapping(
+      nomisAdjustmentId = event.adjustmentId,
+      nomisAdjustmentCategory = "SENTENCE"
+    )?.let {
+      telemetryClient.trackEvent(
+        "sentence-adjustment-updated-synchronisation-success",
+        mapOf(
+          "offenderNo" to event.offenderIdDisplay,
+          "bookingId" to event.bookingId.toString(),
+          "sentenceSequence" to event.sentenceSeq.toString(),
+          "nomisAdjustmentId" to event.adjustmentId.toString(),
+          "adjustmentCategory" to "SENTENCE",
+          "adjustmentId" to it.adjustmentId,
+        ),
+        null
+      )
+    } ?: let {
+      telemetryClient.trackEvent(
+        "sentence-adjustment-created-synchronisation-success",
```

Comment:
Should this have a different event name to above - fail?

Reply:
one is created and one is updated